app/utils/message_data.py

```diff
@@ -1,49 +1,52 @@
 from io import BytesIO
-from typing import TYPE_CHECKING, NamedTuple
+from typing import Self
 
 import discord
 
-if TYPE_CHECKING:
-    import datetime as dt
-
 MAX_ATTACHMENT_SIZE = 67_108_864  # 64 MiB
 
 
-class MessageData(NamedTuple):
-    content: str
-    channel: discord.abc.MessageableChannel
-    created_at: dt.datetime
-    edited_at: dt.datetime | None
-    attachments: list[discord.File]
+class MessageData(discord.Message):
+    files: list[discord.File]
     skipped_attachments: int
-    reactions: list[discord.Reaction]
-
-
-async def scrape_message_data(message: discord.Message) -> MessageData:
-    return MessageData(
-        message.content,
-        message.channel,
-        message.created_at,
-        message.edited_at,
-        *await _get_attachments(message),
-        message.reactions,
-    )
-
-
-async def _get_attachments(message: discord.Message) -> tuple[list[discord.File], int]:
-    if not message.attachments:
-        return [], 0
-
-    attachments: list[discord.File] = []
-    skipped_attachments = 0
-    for attachment in message.attachments:
-        if attachment.size > MAX_ATTACHMENT_SIZE:
-            skipped_attachments += 1
-            continue
-
-        fp = BytesIO(await attachment.read())
-        attachments.append(discord.File(fp, filename=attachment.filename))
 
-    return attachments, skipped_attachments
+    def __init__(self, message: discord.Message) -> None:
+        # This code cannot be in the scrape() constructor itself as the
+        # subclass' __init__() *should not* be called, and overriding the
+        # __init__() is the most elegant solution.
+        for attr in dir(message):
+            val = getattr(type(self), attr)
+            if (
+                # Don't break the class.
+                attr == "__class__"
+                # Acquired already by the subclass declaration.
+                or isinstance(val, property)
+                or callable(val)
+            ):
+                continue
+            with suppress(AttributeError):
+                # At the time of writing, the only things which cause an AttributeError
+                # to be thrown are `call` and everything that starts with `_cs_`.
+                setattr(self, attr, getattr(message, attr))
+
+    @classmethod
+    async def scrape(cls, message: discord.Message) -> Self:
+        # This code cannot go in __init__ as it is async.
+        msg_data = cls(message)
+        await msg_data._get_files(message.attachments)
+        return msg_data
+
+    async def _get_files(self, attachments: list[discord.Attachment]) -> None:
+        self.files: list[discord.File] = []
+        self.skipped_attachments = 0
+        for file in attachments:
+            if file.size > MAX_ATTACHMENT_SIZE:
+                self.skipped_attachments += 1
+                continue
+            fp = BytesIO(await file.read())
+            self.files.append(discord.File(fp, filename=file.filename))
```

Comment:
I'm not sure I like the repeated updates to instance attributes. It's probably ok for our use, but it can generally lead to unexpected things when dealing with fallible functions, so I'd lean towards doing something like:
```suggestion
        files: list[discord.File] = []
        skipped_attachments = 0

        for file in attachments:
            if file.size > MAX_ATTACHMENT_SIZE:
                skipped_attachments += 1
                continue
            fp = BytesIO(await file.read())
            files.append(discord.File(fp, filename=file.filename))

        self.files = files
        self.skipped_attachments = attachments
```
It'll take two more lines but I'll personally feel safer with this (+ I think there's a micro perf win here lol)